zipkin-receiver-scribe/src/main/scala/com/twitter/zipkin/receiver/scribe/ScribeSpanReceiver.scala

```diff
@@ -111,11 +112,19 @@ class ScribeReceiver(
           batchesProcessedStat.add(spans.size)
           ok
         case Throw(NonFatal(e)) =>
-          if (!e.isInstanceOf[QueueFullException])
-            log.warning("Exception in process(): %s".format(e.getMessage))
-          errorStats.counter(e.getClass.getName).incr()
-          pushbackCounter.incr()
-          tryLater
+          // It is not an error if the scribe client decided to cancel its request.
+          // See Finagle FAQ's first entry http://twitter.github.io/finagle/guide/FAQ.html
+          if (e.isInstanceOf[CancellationException] && e.getCause.isInstanceOf[CancelledRequestException]) {
```

Comment:
That's a tricky one... we could try

```scala
(e, e.getCause) match {
  case (_:CancellationException, _:CancelledRequestException) => ok
  case (_:QueueFullException, _) => pushbackCounter.incr(); tryLater
  case _ => ...
```

or

```scala
e match {
  case ex: CancellationException if ex.getCause.isInstanceOf[CancelledRequestException] => ok
  case ex: QueueFullException =>  => pushbackCounter.incr(); tryLater
  case _ => ...
```

but you're right, it's not so easy to express with pattern matching as one would think...